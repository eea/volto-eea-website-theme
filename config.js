import navigationSettings from './src/reducers/navigation/navigation';

export default function applyConfig(config) {
  config.addonReducers = {
    ...config.addonReducers,
    navigationSettings,
  };
  return config;
}