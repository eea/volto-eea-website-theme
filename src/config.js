import biseLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/bise.svg';
import energyLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/energy.svg';
import insituLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/insitu.svg';
import freshwaterLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/freshwater.svg';
import fiseLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/fise.svg';
import ccaLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/cca.svg';
import copernicusLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/copernicus.svg';
import industryLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/industry.svg';
import marineLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/marine.svg';
import eionetLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/eionet.svg';
import eeaLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/images/Header/eea-logo.svg';
import eeaWhiteLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/eea-white.svg';
import climateLogo from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/logo/climate-health.svg';

// TODO: to be consolidated with headerLinks
// Footer.jsx config options
export const footerOpts = {
  header: 'Thematic information platforms',
  logosHeader: 'Managed by',
  contactHeader: 'Contact Us',
  actions: [
    {
      title: 'Privacy',
      link: '/privacy',
    },
    {
      link: '/sitemap',
      title: 'Sitemap',
    },
    {
      link: '/login',
      title: 'CMS Login',
    },
  ],
  copyright: [
    {
      link: '/copyright',
      site: 'EEA',
    },
  ],
  sites: [
    {
      link: 'https://water.europa.eu/marine',
      src: marineLogo,
      alt: 'WISE marine',
    },
    {
      link: 'https://water.europa.eu/freshwater',
      src: freshwaterLogo,
      alt: 'WISE freshwater',
    },
    {
      link: 'https://biodiversity.europa.eu/',
      src: biseLogo,
      alt: 'Biodiversity',
    },
    {
      link: 'https://forest.eea.europa.eu/',
      src: fiseLogo,
      alt: 'Forest information system for europe',
    },
    {
      link: 'https://climate-adapt.eea.europa.eu/observatory',
      src: climateLogo,
      alt: 'European Climate and Health Observatory',
    },
    {
      link: 'https://climate-adapt.eea.europa.eu/',
      src: ccaLogo,
      alt: 'Climate adapt',
    },
    {
      link: 'https://industry.eea.europa.eu/',
      src: industryLogo,
      alt: 'European industrial emissions portal',
    },
    {
      link: 'https://climate-energy.eea.europa.eu/',
      src: energyLogo,
      alt: 'Climate and energy in the EU',
    },
    {
      link: 'https://land.copernicus.eu/',
      src: copernicusLogo,
      alt: 'Copernicus land monitoring service',
    },
    {
      link: 'https://insitu.copernicus.eu/',
      src: insituLogo,
      alt: 'Copernicus in situ',
    },
  ],
  managedBy: [
    {
      link: 'https://www.eea.europa.eu/',
      src: eeaWhiteLogo,
      alt: 'EEA Logo',
      className: 'site logo',
      columnSize: {
        mobile: 6,
        tablet: 12,
        computer: 4,
      },
    },
    {
      link: 'https://www.eionet.europa.eu/',
      src: eionetLogo,
      alt: 'EIONET Logo',
      className: 'eionet logo',
      columnSize: {
        mobile: 6,
        tablet: 12,
        computer: 4,
      },
    },
  ],
  social: [
    {
      name: 'twitter',
      icon: 'ri-twitter-fill',
      link: 'https://twitter.com/euenvironment',
    },
    {
      name: 'facebook',
      icon: 'ri-facebook-box-fill',
      link: 'https://www.facebook.com/European.Environment.Agency',
    },
    {
      name: 'linkedin',
      icon: 'ri-linkedin-fill',
      link: 'https://www.linkedin.com/company/european-environment-agency',
    },
    {
      name: 'youtube',
      icon: 'ri-youtube-fill',
      link: 'https://www.youtube.com/user/EEAvideos',
    },
    {
      name: 'instagram',
      icon: 'ri-instagram-fill',
      link: 'https://www.instagram.com/ourplanet_eu',
    },
    {
      name: 'rss',
      icon: 'ri-rss-fill',
      link: '/subscription/news-feeds',
    },
  ],
  contacts: [
    {
      icon: 'comment outline',
      text: 'About us',
      link: '/about',
      children: [
        {
          link: '/faq',
          text: 'FAQs',
        },
        {
          link: '/careers',
          text: 'Careers',
        },
      ],
    },
    {
      icon: 'comment outline',
      text: 'Contact us',
      link: '/contact-us',
    },
    {
      icon: 'envelope outline',
      text: 'Sign up to our newsletter',
      link: '/newsletter',
    },
  ],
  address: 'Kongens Nytorv 6 1050 Copenhagen K (+45) 33 36 71 00',
};

// Header.jsx config options
export const headerOpts = {
  logo: eeaLogo,
  logoWhite: eeaWhiteLogo,
  partnerLinks: {
    title: 'Environmental information systems',
    links: [
      {
        title: 'Biodiversity Information System for Europe',
        href: 'https://biodiversity.europa.eu/',
      },
      {
        title: 'Climate Adaptation Platform',
        href: 'https://climate-adapt.eea.europa.eu/',
      },
      {
        title: 'Copernicus in situ component',
        href: 'https://insitu.copernicus.eu/',
      },
      {
        title: 'European Industrial Emissions Portal',
        href: 'https://industry.eea.europa.eu/',
      },
      {
        title: 'Forest Information System for Europe',
        href: 'https://forest.eea.europa.eu/',
      },
      {
        title: 'Information Platform for Chemical Monitoring',
        href: 'https://ipchem.jrc.ec.europa.eu/RDSIdiscovery/ipchem/index.html',
      },
      {
        title: 'Marine Water Information System for Europe',
        href: 'https://water.europa.eu/marine',
      },
      {
        title: 'Fresh Water Information System for Europe',
        href: 'https://water.europa.eu/freshwater',
      },
    ],
  },
};

export const languages = [
  { name: 'Български', code: 'bg' },
  { name: 'čeština', code: 'cs' },
  { name: 'Hrvatski', code: 'hr' },
  { name: 'dansk', code: 'da' },
  { name: 'Nederlands', code: 'nl' },
  { name: 'ελληνικά', code: 'el' },
  { name: 'English', code: 'en' },
  { name: 'eesti', code: 'et' },
  { name: 'Suomi', code: 'fi' },
  { name: 'Français', code: 'fr' },
  { name: 'Deutsch', code: 'de' },
  { name: 'magyar', code: 'hu' },
  { name: 'Íslenska', code: 'is' },
  { name: 'italiano', code: 'it' },
  { name: 'Latviešu', code: 'lv' },
  { name: 'lietuvių', code: 'lt' },
  { name: 'Malti', code: 'mt' },
  { name: 'Norsk', code: 'no' },
  { name: 'polski', code: 'pl' },
  { name: 'Português', code: 'pt' },
  { name: 'Română', code: 'ro' },
  { name: 'slovenčina', code: 'sk' },
  { name: 'Slovenščina', code: 'sl' },
  { name: 'Español', code: 'es' },
  { name: 'Svenska', code: 'sv' },
  { name: 'Türkçe', code: 'tr' },
];

export const defaultLanguage = 'en';

export const websiteTitle = 'Site';
export const organisationName = 'European Environment Agency';
export const logoTargetUrl = '/';

export const headerSearchBox = [
  {
    isDefault: true,
    path: '/en/advanced-search',
    placeholder: 'Search...',
  },
  {
    path: '/en/datahub',
    placeholder: 'Search Datahub...',
    description:
      'Looking for more information? Try searching the full EEA website content',
    buttonTitle: 'Go to full site search',
  },
];

export const colors = [
  // Primary & shades
  '#007B6C',
  '#005248',
  '#007B6C',
  '#00A390',
  '#00CCB4',
  '#00F5D8',
  '#85FFF1',
  '#C8FFF8',
  // Secondary & shades
  '#004B7F',
  '#003052',
  '#004B7F',
  '#006BB8',
  '#0083E0',
  '#0A99FF',
  '#47B3FF',
  '#A0D7FF',
  // Tertiary & shades
  '#2E3E4C',
  '#3D5265',
  '#4C677F',
  '#6989A5',
  '#8EA6C2',
  '#ACCAE5',
  '#DAE8F4',
  // Black & White shades
  '#000000',
  '#323232',
  '#67696b',
  '#BCBEC0',
  '#E6E7E8',
  '#F9F9F9',
  '#FFFFFF',
];
