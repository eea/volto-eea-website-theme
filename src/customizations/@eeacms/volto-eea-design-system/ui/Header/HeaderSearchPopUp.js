import React from 'react';
import { Container, Input } from 'semantic-ui-react';
import { withRouter } from 'react-router-dom';
import { useClickOutside } from '@eeacms/volto-eea-design-system/helpers';
import config from '@plone/volto/registry';

function HeaderSearchPopUp({
  history,
  location,
  onClose,
  searchInputRef,
  triggerRefs = [],
}) {
  const nodeRef = React.useRef();
  const { eea } = config.settings;
  const defaultView = eea.headerSearchBox.filter((v) => v.isDefault);
  const localView = eea.headerSearchBox.filter((v) =>
    location.pathname.includes(v.path),
  );
  const activeView = localView.length > 0 ? localView[0] : defaultView[0];

  const [text, setText] = React.useState('');

  useClickOutside({ targetRefs: [nodeRef, ...triggerRefs], callback: onClose });

  const onChangeText = (event, { value }) => {
    setText(value);
    event.preventDefault();
  };

  const onSubmit = (event) => {
    history.push(`${activeView.path}?q=${text}`);

    if (window?.searchContext?.resetSearch) {
      window.searchContext.resetSearch({ searchTerm: text });
    }

    onClose();
    event.preventDefault();
  };

  return (
    <div id="search-box" ref={nodeRef}>
      <form method="get" onSubmit={onSubmit}>
        <Container>
          <div className="wrapper">
            <Input
              ref={searchInputRef}
              className="search"
              onChange={onChangeText}
              icon={{
                className: 'ri-search-line',
                link: true,
                onClick: onSubmit,
              }}
              placeholder={activeView.placeholder}
              fluid
            />
          </div>
        </Container>
      </form>
      {(activeView.description || activeView.buttonTitle) && (
        <div className="advanced-search">
          <Container>
            <p>{activeView.description}</p>
            <a
              href={defaultView[0].path}
              className="ui button white inverted"
              title="Advanced search"
            >
              {activeView.buttonTitle}
            </a>
          </Container>
        </div>
      )}
    </div>
  );
}

export default withRouter(HeaderSearchPopUp);
