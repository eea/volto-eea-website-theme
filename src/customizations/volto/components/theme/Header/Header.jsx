/**
 * Header component.
 * @module components/theme/Header/Header
 */

import React from 'react';
import { Dropdown, Image } from 'semantic-ui-react';
import { connect, useDispatch, useSelector } from 'react-redux';

import { withRouter } from 'react-router-dom';
import { UniversalLink } from '@plone/volto/components';
import {
  getBaseUrl,
  hasApiExpander,
  flattenToAppURL,
} from '@plone/volto/helpers';
import { getNavigation } from '@plone/volto/actions';
import { Header, Logo } from '@eeacms/volto-eea-design-system/ui';
import { usePrevious } from '@eeacms/volto-eea-design-system/helpers';
import { find } from 'lodash';
import globeIcon from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/images/Header/global-line.svg';
import eeaFlag from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/images/Header/eea.png';

import config from '@plone/volto/registry';
import { compose } from 'recompose';
import { BodyClass } from '@plone/volto/helpers';

import cx from 'classnames';

function removeTrailingSlash(path) {
  return path.replace(/\/+$/, '');
}

/**
 * EEA Specific Header component.
 */
const EEAHeader = ({ pathname, token, items, history, subsite }) => {
  const currentLang = useSelector((state) => state.intl.locale);
  const translations = useSelector(
    (state) => state.content.data?.['@components']?.translations?.items,
  );

  const router_pathname = useSelector((state) => {
    return removeTrailingSlash(state.router?.location?.pathname) || '';
  });

  const isSubsite = subsite?.['@type'] === 'Subsite';

  const isHomePageInverse = useSelector((state) => {
    const layout = state.content?.data?.layout;
    const has_home_layout =
      layout === 'homepage_inverse_view' ||
      (__CLIENT__ && document.body.classList.contains('homepage-inverse'));
    return (
      has_home_layout &&
      (removeTrailingSlash(pathname) === router_pathname ||
        router_pathname.endsWith('/edit'))
    );
  });

  const { eea } = config.settings;
  const headerOpts = eea.headerOpts || {};
  const { logo, logoWhite } = headerOpts || {};
  const width = useSelector((state) => state.screen?.width);
  const dispatch = useDispatch();
  const previousToken = usePrevious(token);
  const [language, setLanguage] = React.useState(
    currentLang || eea.defaultLanguage,
  );

  React.useEffect(() => {
    const { settings } = config;
    const base_url = getBaseUrl(pathname);
    if (!hasApiExpander('navigation', base_url)) {
      dispatch(getNavigation(base_url, settings.navDepth));
    }
  }, [pathname, dispatch]);

  React.useEffect(() => {
    if (token !== previousToken) {
      const { settings } = config;
      const base = getBaseUrl(pathname);
      if (!hasApiExpander('navigation', base)) {
        dispatch(getNavigation(base, settings.navDepth));
      }
    }
  }, [token, dispatch, pathname, previousToken]);

  return (
    <Header menuItems={items}>
      {isHomePageInverse && <BodyClass className="homepage" />}
      <Header.TopHeader>
        <Header.TopItem className="official-union">
          <Image src={eeaFlag} alt="eea flag"></Image>
          <Header.TopDropdownMenu
            text="An official website of the European Union | How do you know?"
            tabletText="EEA information systems"
            mobileText=" "
            icon="chevron down"
            aria-label="dropdown"
            className=""
            viewportWidth={width}
          >
            <div
              className="content"
              role="menu"
              tabIndex="0"
              onClick={(evt) => evt.stopPropagation()}
              onKeyDown={(evt) => evt.stopPropagation()}
            >
              <p>
                All official European Union website addresses are in the{' '}
                <b>europa.eu</b> domain.
              </p>
              <a
                href="https://europa.eu/european-union/contact/institutions-bodies_en"
                target="_blank"
                rel="noreferrer"
                role="option"
                aria-selected="false"
              >
                See all EU institutions and bodies
              </a>
            </div>
          </Header.TopDropdownMenu>
        </Header.TopItem>

        {!!headerOpts.partnerLinks && (
          <Header.TopItem>
            <Header.TopDropdownMenu
              id="theme-sites"
              text={headerOpts.partnerLinks.title}
              viewportWidth={width}
            >
              <div className="wrapper">
                {headerOpts.partnerLinks.links.map((item, index) => (
                  <Dropdown.Item key={index}>
                    <a
                      href={item.href}
                      className="site"
                      target="_blank"
                      rel="noreferrer"
                    >
                      {item.title}
                    </a>
                  </Dropdown.Item>
                ))}
              </div>
            </Header.TopDropdownMenu>
          </Header.TopItem>
        )}

        {config.settings.isMultilingual && (
          <Header.TopDropdownMenu
            id="language-switcher"
            className="item"
            hasLanguageDropdown={config.settings.supportedLanguages.length > 1}
            text={`${language.toUpperCase()}`}
            mobileText={`${language.toUpperCase()}`}
            icon={
              <Image src={globeIcon} alt="language dropdown globe icon"></Image>
            }
            viewportWidth={width}
          >
            <ul
              className="wrapper language-list"
              role="listbox"
              aria-label="language switcher"
            >
              {eea.languages.map((item, index) => (
                <Dropdown.Item
                  as="li"
                  key={index}
                  text={
                    <span>
                      {item.name}
                      <span className="country-code">
                        {item.code.toUpperCase()}
                      </span>
                    </span>
                  }
                  onClick={() => {
                    const translation = find(translations, {
                      language: item.code,
                    });
                    const to = translation
                      ? flattenToAppURL(translation['@id'])
                      : `/${item.code}`;
                    setLanguage(item.code);
                    history.push(to);
                  }}
                ></Dropdown.Item>
              ))}
            </ul>
          </Header.TopDropdownMenu>
        )}
      </Header.TopHeader>
      <Header.Main
        pathname={pathname}
        inverted={isHomePageInverse ? true : false}
        transparency={isHomePageInverse ? true : false}
        logo={
          <div {...(isSubsite ? { className: 'logo-wrapper' } : {})}>
            <Logo
              src={isHomePageInverse ? logoWhite : logo}
              title={eea.websiteTitle}
              alt={eea.organisationName}
              url={eea.logoTargetUrl}
            />

            {!!subsite && subsite.title && (
              <UniversalLink item={subsite} className="subsite-logo">
                {subsite.title}
              </UniversalLink>
            )}
          </div>
        }
        menuItems={items}
        renderGlobalMenuItem={(item, { onClick }) => (
          <a
            href={item.url || '/'}
            title={item.title}
            onClick={(e) => {
              e.preventDefault();
              onClick(e, item);
            }}
          >
            {item.title}
          </a>
        )}
        renderMenuItem={(item, options, props) => (
          <UniversalLink
            href={item.url || '/'}
            title={item.nav_title || item.title}
            {...(options || {})}
            className={cx(options?.className, {
              active: item.url === router_pathname,
            })}
          >
            {props?.iconPosition !== 'right' && props?.children}
            <span>{item.nav_title || item.title}</span>
            {props?.iconPosition === 'right' && props?.children}
          </UniversalLink>
        )}
      ></Header.Main>
    </Header>
  );
};

export default compose(
  withRouter,
  connect(
    (state) => ({
      token: state.userSession.token,
      items: state.navigation.items,
      subsite: state.content.data?.['@components']?.subsite,
    }),
    { getNavigation },
  ),
)(EEAHeader);
