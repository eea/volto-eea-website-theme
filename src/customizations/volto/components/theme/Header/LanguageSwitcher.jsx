import React from 'react';
import { useSelector } from 'react-redux';
import { Dropdown, Image } from 'semantic-ui-react';
import { flattenToAppURL } from '@plone/volto/helpers';
import { find } from 'lodash';
import globeIcon from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/images/Header/global-line.svg';
import config from '@plone/volto/registry';
import { Header } from '@eeacms/volto-eea-design-system/ui';

/**
 * LanguageSwitcher component.
 * Provides a dropdown menu for language selection, changing the application's
 * language and navigating to the corresponding translated URL.
 *
 * @param {Object} props - The component props.
 * @param {number} props.width - The viewport width to adjust the dropdown display.
 * @param {Object} props.history - The history object from React Router for navigation.
 */
const LanguageSwitcher = ({ width, history }) => {
  const currentLang = useSelector((state) => state.intl.locale);
  const translations = useSelector(
    (state) => state.content.data?.['@components']?.translations?.items,
  );
  const { eea } = config.settings;

  const [language, setLanguage] = React.useState(
    currentLang || eea.defaultLanguage,
  );

  return (
    <Header.TopDropdownMenu
      id="language-switcher"
      className="item"
      text={`${language.toUpperCase()}`}
      mobileText={`${language.toUpperCase()}`}
      icon={<Image src={globeIcon} alt="language dropdown globe icon"></Image>}
      viewportWidth={width}
    >
      <ul
        className="wrapper language-list"
        role="listbox"
        aria-label="language switcher"
      >
        {eea.languages.map((item, index) => (
          <Dropdown.Item
            as="li"
            key={index}
            text={
              <span>
                {item.name}
                <span className="country-code">{item.code.toUpperCase()}</span>
              </span>
            }
            onClick={() => {
              const translation = find(translations, {
                language: item.code,
              });
              const to = translation
                ? flattenToAppURL(translation['@id'])
                : `/${item.code}`;
              setLanguage(item.code);
              history.push(to);
            }}
          ></Dropdown.Item>
        ))}
      </ul>
    </Header.TopDropdownMenu>
  );
};

export default LanguageSwitcher;
