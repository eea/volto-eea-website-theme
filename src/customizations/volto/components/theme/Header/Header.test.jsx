import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';

import Header from './Header';

const mockStore = configureStore();
let history = createMemoryHistory();

describe('Header', () => {
  it('renders a header component', () => {
    const store = mockStore({
      userSession: { token: null },
      intl: {
        locale: 'en',
        messages: {},
      },
      navigation: {
        items: ['en'],
      },
      content: {
        data: {
          layout: 'homepage_inverse_view',
        },
      },
      router: {
        location: {
          pathname: '/home/',
        },
      },
    });

    config.settings = {
      ...config.settings,
      eea: {
        headerOpts: undefined,
      },
    };

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Header pathname="/home" />
        </Router>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a header component', () => {
    const store = mockStore({
      userSession: { token: null },
      intl: {
        locale: 'en',
        messages: {},
      },
      navigation: {
        items: ['en'],
      },
      content: {
        data: {
          layout: 'homepage_inverse_view',
        },
      },
      router: {
        location: {
          pathname: '/home/',
        },
      },
    });

    config.settings = {
      ...config.settings,
      eea: {
        headerOpts: {},
      },
    };

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Header pathname="/blog" />
        </Router>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a header component', () => {
    const store = mockStore({
      userSession: { token: null },
      intl: {
        locale: undefined,
        messages: {},
      },
      navigation: {
        items: ['en'],
      },
      content: {
        data: {
          layout: 'homepage_view',
          '@components': {
            translations: {
              items: [{ language: 'en' }, { language: 'ro' }],
            },
          },
        },
      },
      router: {
        location: {
          pathname: '/home/',
        },
      },
    });

    config.settings = {
      ...config.settings,
      eea: {
        headerOpts: {
          partnerLinks: {
            links: [{ href: '/link1', title: 'link 1' }],
          },
        },
        defaultLanguage: 'en',
        languages: [{ code: 'en' }, { code: 'ro' }],
      },
      isMultilingual: true,
      supportedLanguages: ['en', 'ro'],
      hasLanguageDropdown: true,
    };

    const { container, rerender } = render(
      <Provider store={store}>
        <Router history={history}>
          <Header pathname="/blog" />
        </Router>
      </Provider>,
    );

    fireEvent.click(container.querySelector('.content'));
    fireEvent.keyDown(container.querySelector('.content'), { keyCode: 37 });
    fireEvent.keyDown(container.querySelector('.content a'), { keyCode: 37 });
    fireEvent.keyDown(container.querySelector('a[href="/link1"]'), {
      keyCode: 37,
    });
    fireEvent.click(container.querySelector('.country-code'));

    // expect(getByText('da')).toBeInTheDocument();

    rerender(
      <Provider store={{ ...store, userSession: { token: '1234' } }}>
        <Router history={history}>
          <Header pathname="/blog" />
        </Router>
      </Provider>,
    );
  });

  it('renders a header component', () => {
    const store = mockStore({
      userSession: { token: null },
      intl: {
        locale: undefined,
        messages: {},
      },
      navigation: {
        items: ['en'],
      },
      content: {
        data: {
          layout: 'homepage_view',
          '@components': {
            subsite: {
              '@type': 'Subsite',
              title: 'Home Page',
              subsite_logo: {
                scales: {
                  mini: {
                    download:
                      'http://localhost:8080/Plone/subsite_logo/@@images/image/mini',
                  },
                },
              },
            },
            translations: {
              items: [{ language: 'ro' }],
            },
          },
        },
      },
      router: {
        location: {
          pathname: '/home/',
        },
      },
    });

    config.settings = {
      ...config.settings,
      eea: {
        headerOpts: {
          partnerLinks: {
            links: [{ href: '/link1', title: 'link 1' }],
          },
        },
        defaultLanguage: 'en',
        languages: [{ code: 'en' }, { code: 'ro' }],
      },
      isMultilingual: true,
      supportedLanguages: ['en', 'ro'],
      hasLanguageDropdown: true,
    };

    const { container, rerender } = render(
      <Provider store={store}>
        <Router history={history}>
          <Header pathname="/blog" />
        </Router>
      </Provider>,
    );

    fireEvent.click(container.querySelector('.content'));
    fireEvent.keyDown(container.querySelector('.content'), { keyCode: 37 });
    fireEvent.keyDown(container.querySelector('.content a'), { keyCode: 37 });
    fireEvent.keyDown(container.querySelector('a[href="/link1"]'), {
      keyCode: 37,
    });
    fireEvent.click(container.querySelector('.country-code'));

    // expect(getByText('da')).toBeInTheDocument();

    rerender(
      <Provider store={{ ...store, userSession: { token: '1234' } }}>
        <Router history={history}>
          <Header pathname="/blog" />
        </Router>
      </Provider>,
    );
  });

  it('renders a header component', () => {
    const store = mockStore({
      userSession: { token: null },
      intl: {
        locale: undefined,
        messages: {},
      },
      navigation: {
        items: [
          { url: '/test1', title: 'test 1', nav_title: 'Test 1', items: [] },
          { url: undefined, title: 'test 2', items: [] },
        ],
      },
      content: {
        data: {
          layout: 'homepage_view',
          '@components': {
            subsite: {
              '@type': 'Subsite',
              title: 'Home Page',
              subsite_logo: undefined,
            },
            translations: {
              items: [{ language: 'ro' }],
            },
          },
        },
      },
      router: {
        location: {
          pathname: '/home/',
        },
      },
    });

    config.settings = {
      ...config.settings,
      eea: {
        headerOpts: {
          partnerLinks: {
            links: [{ href: '/link1', title: 'link 1' }],
          },
        },
        defaultLanguage: 'en',
        languages: [{ code: 'en' }, { code: 'ro' }],
      },
      isMultilingual: true,
      supportedLanguages: ['en', 'ro'],
      hasLanguageDropdown: true,
    };

    const { container, rerender } = render(
      <Provider store={store}>
        <Router history={history}>
          <Header pathname="/blog" />
        </Router>
      </Provider>,
    );

    fireEvent.click(container.querySelector('.content'));
    fireEvent.keyDown(container.querySelector('.content'), { keyCode: 37 });
    fireEvent.keyDown(container.querySelector('.content a'), { keyCode: 37 });
    fireEvent.keyDown(container.querySelector('a[href="/link1"]'), {
      keyCode: 37,
    });
    fireEvent.click(container.querySelector('.country-code'));
    fireEvent.click(container.querySelector('a[href="/test1"]'));

    // expect(getByText('da')).toBeInTheDocument();

    rerender(
      <Provider store={{ ...store, userSession: { token: '1234' } }}>
        <Router history={history}>
          <Header pathname="/blog" />
        </Router>
      </Provider>,
    );
  });
});
