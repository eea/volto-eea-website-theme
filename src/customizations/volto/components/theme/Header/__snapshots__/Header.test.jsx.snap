// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders a header component 1`] = `
<div
  className="eea header"
>
  <div
    className="top bar"
  >
    <div
      className="ui container"
    >
      <div
        className="item header-top-item official-union"
      >
        <img
          alt="European Union flag"
          className="ui image"
          src="eea.png"
        />
        <div
          aria-expanded={false}
          aria-label="dropdown"
          className="ui dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-atomic={true}
            aria-live="polite"
            className="divider text"
            role="alert"
          >
            An official website of the European Union | How do you know?
          </div>
          <i
            aria-hidden="true"
            className="chevron down icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
            role="option"
          >
            <div
              className="content"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              tabIndex={0}
            >
              <p>
                All official European Union website addresses are in the
                 
                <b>
                  europa.eu
                </b>
                 domain.
              </p>
              <a
                href="https://europa.eu/european-union/contact/institutions-bodies_en"
                onKeyDown={[Function]}
                rel="noreferrer"
                target="_blank"
              >
                See all EU institutions and bodies
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="main bar transparency"
  >
    <div
      className="ui container"
    >
      <div
        className="ui grid"
      >
        <div
          className="four wide computer eight wide mobile eight wide tablet column"
        >
          <div>
            <a
              className="logo"
              href=""
              onClick={[Function]}
            >
              <img
                className="ui image eea-logo"
              />
            </a>
          </div>
        </div>
        <div
          className="eight wide computer four wide mobile four wide tablet column"
        >
          <div
            className="main-menu inverted"
          >
            <ul
              className="ui text eea-main-menu tablet or lower hidden menu"
              id="navigation"
            >
              <li
                className="item"
                onClick={[Function]}
              >
                <a
                  href="/"
                  onClick={[Function]}
                />
              </li>
            </ul>
            <button
              aria-haspopup="true"
              aria-pressed="false"
              className="search-action"
              onClick={[Function]}
              tabIndex="0"
            >
              <img
                alt="search button open/close"
                className="ui image"
                src="search-line.svg"
              />
            </button>
            <button
              aria-haspopup="true"
              aria-pressed="false"
              className="burger-action mobile "
              onClick={[Function]}
              tabIndex="0"
            >
              <img
                alt="menu icon open/close"
                className="ui image"
                src="menu-line.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header renders a header component 2`] = `
<div
  className="eea header"
>
  <div
    className="top bar"
  >
    <div
      className="ui container"
    >
      <div
        className="item header-top-item official-union"
      >
        <img
          alt="European Union flag"
          className="ui image"
          src="eea.png"
        />
        <div
          aria-expanded={false}
          aria-label="dropdown"
          className="ui dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-atomic={true}
            aria-live="polite"
            className="divider text"
            role="alert"
          >
            An official website of the European Union | How do you know?
          </div>
          <i
            aria-hidden="true"
            className="chevron down icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
            role="option"
          >
            <div
              className="content"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              tabIndex={0}
            >
              <p>
                All official European Union website addresses are in the
                 
                <b>
                  europa.eu
                </b>
                 domain.
              </p>
              <a
                href="https://europa.eu/european-union/contact/institutions-bodies_en"
                onKeyDown={[Function]}
                rel="noreferrer"
                target="_blank"
              >
                See all EU institutions and bodies
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="main bar "
  >
    <div
      className="ui container"
    >
      <div
        className="ui grid"
      >
        <div
          className="four wide computer eight wide mobile eight wide tablet column"
        >
          <div>
            <a
              className="logo"
              href=""
              onClick={[Function]}
            >
              <img
                className="ui image eea-logo"
              />
            </a>
          </div>
        </div>
        <div
          className="eight wide computer four wide mobile four wide tablet column"
        >
          <div
            className="main-menu"
          >
            <ul
              className="ui text eea-main-menu tablet or lower hidden menu"
              id="navigation"
            >
              <li
                className="item"
                onClick={[Function]}
              >
                <a
                  href="/"
                  onClick={[Function]}
                />
              </li>
            </ul>
            <button
              aria-haspopup="true"
              aria-pressed="false"
              className="search-action"
              onClick={[Function]}
              tabIndex="0"
            >
              <img
                alt="search button open/close"
                className="ui image"
                src="search-line.svg"
              />
            </button>
            <button
              aria-haspopup="true"
              aria-pressed="false"
              className="burger-action mobile "
              onClick={[Function]}
              tabIndex="0"
            >
              <img
                alt="menu icon open/close"
                className="ui image"
                src="menu-line.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
